### PR TITLE
SHOR-35: Add missing screens for find and merge duplicate contacts and fixing cases timeout issue for manage groups.

### PR DESCRIPTION
**Pull request by swastikpareek**
## Overview
This PR covers the following screens
* Find and Merge Duplicate Contacts - Edit Household Rule  - results
  77041666_find_and_merge_duplicate_contacts_-_edit_household_rule__-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Edit Individual Rule  - results
  77041666_find_and_merge_duplicate_contacts_-_edit_individual_rule__-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Edit Organisation Rule  - results
  77041666_find_and_merge_duplicate_contacts_-_edit_organisation_rule__-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Use Rule - results
  77041666_find_and_merge_duplicate_contacts_-_use_rule_-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Use Rule - Show duplicate contacts - results
  77041666_find_and_merge_duplicate_contacts_-_use_rule_-_show_duplicate_contacts_-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Use Rule - Merge duplicate contacts - results
  77041666_find_and_merge_duplicate_contacts_-_use_rule_-_merge_duplicate_contacts_-_results_0_document_0_desktop
* Find and Merge Duplicate Contacts - Use Rule - Not a Duplicate Modal - results
  77041666_find_and_merge_duplicate_contacts_-_use_rule_-_not_a_duplicate_modal_-_results_0_ui-dialog_0_desktop

#### Problem
Initially it closes the first notification it finds and was timing out if page has multiple notifications
#### Fix
  Iterating the code till all notification windows are closed fixed the issue. 
  See backstop_data/engine_scripts/puppet/page-objects/crm-page.js.
#### Current issue
Some scenarios for manage groups were throwing navigation time out randomly too much. After debugging more I found out that there was a race condition between browser page load and awaiting response from the `engine.click` event. So in layman terms the browsers loads to the next page till the Promise for engine.click resolves and so waitForNavigation always start on the new page and waits forever as the page is already loaded.
#### Fix
Starting both the Promise asynchronously did the magic. So `Promise.all` starts both the processes asynchronously and waits till both of them finishes allowing waitForNavigation to listen for the url change without any delays.
To make sure every scenario uses this process. *waitForLinkNavigation()* function is created in page Object.
See backstop_data/engine_scripts/puppet/page-objects/crm-page.js

### CODE REVIEW

File: backstop_data/engine_scripts/puppet/contact/find-and-merge-duplicate-contacts.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = async (engine, scenario, vp) => {
+  await require('./find-and-merge-show-duplicates')(engine, scenario, vp);
+  await engine.waitForSelector(".crm-dedupe-flip + a");
+  await engine.click(".crm-dedupe-flip + a");
+};
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/contact/find-and-merge-duplicate-contacts.js]: make sure your IDE leaves an empty line on save

File: backstop_data/engine_scripts/puppet/contact/find-and-merge-edit-rule-household.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+'use strict';
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/contact/find-and-merge-edit-rule-household.js]: since all this files have the `find-and-merge` prefix, wouldn't it make sense to have a `contact/find-and-merge/` folder? what do you think

File: backstop_data/engine_scripts/puppet/contact/find-and-merge-show-duplicates.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = async (engine, scenario, vp) => {
+  await require('./find-and-merge-use-rule')(engine, scenario, vp);
+  await engine.waitForSelector("#_qf_DedupeFind_next-bottom", {visible: true});
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/contact/find-and-merge-show-duplicates.js]: check your syntax, it's not 
```js
{visible: true}
```
but
```js
{ visible: true }
```

File: backstop_data/engine_scripts/puppet/page-objects/crm-page.js [hunk 4]
```diff
@@ -42,11 +42,14 @@ module.exports = class CrmPage {
    * Closes all active notifications.
    */
   async closeErrorNotifications () {
-    const exists = !!(await this.engine.$('.ui-notify-message'));
-
-    if (exists) {
-      await this.engine.click('a.ui-notify-cross.ui-notify-close');
-      await this.engine.waitFor('.ui-notify-message', { hidden: true });
+    while(1) {
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/page-objects/crm-page.js]: let's not have infinite loops, ever, please

File: backstop_data/engine_scripts/puppet/page-objects/crm-page.js [hunk 5]
```diff
@@ -42,11 +42,14 @@ module.exports = class CrmPage {
...
+      const exists = !!(await this.engine.$('.ui-notify-message'));
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/page-objects/crm-page.js]: if the problem was that there are multiple notifications, then puppeteer provides the [`.$$`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageselector-1) method, which is the equivalent of `document.querySelectorAll()`.

You could use it to know how many notifications are there to be closed, and then set up a loop to close as many notifications as `.$$` found. I'd suggest to process the list in sequence (see [here](https://blog.lavrton.com/javascript-loops-how-to-handle-async-await-6252dd3c795), to make sure we have a reliable result

```js
const notifications = this.engine.$$('.ui-notify-message');

if (notifications.length) {
  for (const notification of notifications) {
    await this.engine.click('a.ui-notify-cross.ui-notify-close');
    await this.engine.waitFor('.ui-notify-message', { hidden: true });
  }
}
```
Notice that we don't really use the `notification` object in this case, as we're using the `notifications` array only to know *how many* notifications there are to close, not *which one* we are closing at any given time

File: backstop_data/engine_scripts/puppet/contact/find-and-merge-edit-rule-household.js (hunk 2 above)
Comment (swastikpareek) [reply backstop_data/engine_scripts/puppet/contact/find-and-merge-edit-rule-household.js]: I think thats a nice idea! Let me do it.

File: backstop_data/engine_scripts/puppet/contact/find-and-merge-duplicate-contacts.js (hunk 1 above)
Comment (swastikpareek) [reply backstop_data/engine_scripts/puppet/contact/find-and-merge-duplicate-contacts.js]: Done!
